File: src/c/_0896_Monotonic_Array.c
bool isMonotonic(int *nums, int numsSize)
{
    if (nums[0] < nums[numsSize - 1])
    {
        for (int i = 1; i < numsSize; i++)
            if (nums[i - 1] > nums[i])
                return false;
    }
    else
    {
        for (int i = 1; i < numsSize; i++)
            if (nums[i - 1] < nums[i])
                return false;
    }

    return true;
}

File: src/python/_0896_Monotonic_Array.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        isIncreasing = False
        isDecreasing = False

        for i in range(1, len(nums)):
            if nums[i - 1] == nums[i]:
                continue
            elif nums[i - 1] < nums[i]:
                isIncreasing = True
            else:
                isDecreasing = True

            if isIncreasing and isDecreasing:
                return False

        return True
